src/lambda/commands/uploadLambda.ts

```diff
@@ -197,37 +197,40 @@ function createConfirmDeploymentPrompter(lambda: LambdaFunction) {
     })()
 }
 
-interface UploadLambdaWizardState {
+export interface UploadLambdaWizardState {
     readonly uploadType: 'zip' | 'directory'
     readonly targetUri: vscode.Uri
     readonly directoryBuildType: 'zip' | 'sam'
     readonly confirmedDeploy: boolean
     readonly lambda: LambdaFunction
 }
 
-class UploadLambdaWizard extends Wizard<UploadLambdaWizardState> {
+export class UploadLambdaWizard extends Wizard<UploadLambdaWizardState> {
     constructor(lambda?: LambdaFunction, invokePath?: vscode.Uri) {
         super({ initState: { lambda } })
         this.form.lambda.region.bindPrompter(() => createRegionPrompter().transform(region => region.id))
 
         this.form.uploadType.bindPrompter(() => createUploadTypePrompter())
```

Comment:
We should be skipping the upload type when the target is a directory. So we can just move this line to the first `else` block and set a default value for `directory` otherwise.

Reply:
Thanks, fixed and updated tests.